### PR TITLE
Remove unsupported series from core

### PR DESCRIPTION
**Pull request by jack-w-shaw**
As discussed in the go around yesterday, 

Most notably, drop OpenSUSE and MacOSX entirely, which despite being listed as supported series, are not supported targets and never have been (they are supported clients)

Also drop EoL CentOS and Ubuntu versions. Keep Centos7 for now, which is EoL in June 2024

Also, as a flyby, make some small changes to test_deploy_os. Note: this test is broken at the moment, and still doesn't pass

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

All unit tests pass

```
juju bootstrap lxd lxd
juju add-model m
juju deploy ubuntu
```

```
$ juju add-machine --series opensuseleap
ERROR attempting to convert "opensuseleap" to a base: series "opensuseleap" not valid
```

On a MacOS device:
```
make install
juju bootstrap aws/eu-west-2 aws --bootsrap-constraints arch=arm64 # For M1 devices
```

### CODE REVIEW

File: core/os/os.go [hunk 1]
```diff
@@ -26,6 +27,8 @@ func (t OSType) String() string {
 		return "Ubuntu"
 	case Windows:
 		return "Windows"
```
Comment (SimonRichardson) [inline core/os/os.go:29]: Whilst you're here, can you remove windows.

**Review by SimonRichardson — COMMENTED** [review]
> On a MacOS device:
> `make install`

It isn't enough to just run `make install`. You'll need to either get hold of a mac machine to run this against, or ask someone nicely to run some tests for you.

**Comment by SimonRichardson** [comment]
I've added @hmlanigan and @tlm as they're macOS drivers.

File: core/os/os.go (hunk 1 above)
Comment (jack-w-shaw) [reply core/os/os.go:29]: I don't think we can unfortunately

**Review by hmlanigan — COMMENTED** [review]
I can build the tip of 3.1 on macOS and bootstrap was with agent 3.1.5 which is released. However if I a client built from this pr to do the same, it panics.

https://pastebin.canonical.com/p/jHkBRztyBT/

When I built from scratch a second time, it failed with:
```
# github.com/juju/juju/apiserver/facades/client/secrets
<autogenerated>:1: internal compiler error: panic: interface conversion: types.Object is nil, not *ir.Name

Please file a bug report including a short program that triggers the error.
https://go.dev/issue/new
github.com/juju/juju/apiserver/facades/client/usermanager
```

I'm using go 1.20.2. Trying to compile again, it succeeded and bootstrap worked.

There are many tests with "opensuseleap" patched into them still, as well as ssh_provisioner.go

@tlm How are you doing using this change? My desktop has been unhappy all week.

File: core/base/supported.go [hunk 2]
```diff
@@ -393,24 +390,6 @@ var centosSeries = map[SeriesName]seriesVersion{
 		Version:      "7",
 		Supported:    true,
 	},
-	Centos8: {
```
Comment (tlm) [inline core/base/supported.go:396]: I am confused. We are removing Centos 8 & 9 but not 7 ?

File: cloudconfig/machinecloudconfig.go [hunk 3]
```diff
@@ -81,7 +81,7 @@ func NewMachineInitReaderFromConfig(cfg MachineInitReaderConfig) InitReader {
 // machine. It is sourced from both Cloud-Init and Curtin data.
 func (r *MachineInitReader) GetInitConfig() (map[string]interface{}, error) {
 	switch utilsos.OSTypeForName(r.config.Base.OS) {
-	case utilsos.Ubuntu, utilsos.CentOS, utilsos.OpenSUSE:
+	case utilsos.Ubuntu, utilsos.CentOS:
```
Comment (tlm) [inline cloudconfig/machinecloudconfig.go:84]: Unsure if we are removing Centos entirely as 7 still seems to be supported but should we remove it here if it's a full rip out?

File: provider/common/disk.go [hunk 4]
```diff
@@ -13,7 +13,7 @@ import (
 // data.
 func MinRootDiskSizeGiB(os jujuos.OSType) uint64 {
 	switch os {
-	case jujuos.Ubuntu, jujuos.CentOS, jujuos.OpenSUSE:
+	case jujuos.Ubuntu, jujuos.CentOS:
```
Comment (tlm) [inline provider/common/disk.go:16]: Seems like there is a quick win here to get rid of all this code and just `return 8` and also `_ jujuos.OSType`

File: version/current_test.go [hunk 5]
```diff
@@ -28,27 +27,15 @@ func (*CurrentSuite) TestCurrentSeries(c *gc.C) {
...
-		}
-	} else {
-		//OpenSUSE lsb-release returns n/a
 		currentOS, err := corebase.GetOSFromSeries(s)
 		c.Assert(err, gc.IsNil)
-		if string(out) == "n/a" && currentOS == os.OpenSUSE {
-			c.Check(s, gc.Matches, "opensuseleap")
-		} else {
-			c.Assert(string(out), gc.Equals, "Codename:\t"+s+"\n")
+		if s != "n/a" {
```
Comment (tlm) [inline version/current_test.go:32]: Does it make sense to just combine all these if statements together now?

**Review by tlm — APPROVED** [review]
lgtm

The following steps worked for me in testing.

```bash
AGENT_PACKAGE_PLATFORMS=linux/amd64 make simplestreams install
juju bootstrap --bootstrap-constraints arch=amd64 aws/ap-southeast-2
juju add-model test
juju add-machine
```

File: core/base/supported.go (hunk 2 above)
Comment (jack-w-shaw) [reply core/base/supported.go:396]: Yes. CentOS 7 is still supported by 8 and 9 are EoL

File: cloudconfig/machinecloudconfig.go (hunk 3 above)
Comment (jack-w-shaw) [reply cloudconfig/machinecloudconfig.go:84]: We are keeping CentOS 7 so we need to keep this here

File: core/base/supported.go (hunk 2 above)
Comment (SimonRichardson) [reply core/base/supported.go:396]: Centos9 is a rolling update, so can't be EOL.

File: core/base/supported.go [hunk 6]
```diff
@@ -393,24 +390,6 @@ var centosSeries = map[SeriesName]seriesVersion{
...
-		WorkloadType: OtherWorkloadType,
-		Version:      "8",
-		Supported:    true,
-	},
-	Centos9: {
-		WorkloadType: OtherWorkloadType,
-		Version:      "9",
-		Supported:    true,
-	},
```
Comment (SimonRichardson) [inline core/base/supported.go:405]: Centos9 is a rolling update, so can't be EOL, we need to put this one back in.

**Review by SimonRichardson — CHANGES_REQUESTED** [review]

File: version/current_test.go (hunk 5 above)
Comment (jack-w-shaw) [reply version/current_test.go]: Done

File: core/base/supported.go (hunk 6 above)
Comment (jack-w-shaw) [reply core/base/supported.go]: Centos9 has been restored :cry:

**Review by SimonRichardson — APPROVED** [review]
As we only ever supported opensuse 42, which is EOL it's good to get rid of this code.

**Pull request merged**